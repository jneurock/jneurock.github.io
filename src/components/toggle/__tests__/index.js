import React from 'react';
import Toggle from '../index';
import renderer from 'react-test-renderer';
import { afterEach, describe, expect, it, jest } from '@jest/globals';
import { cleanup, fireEvent, render } from '@testing-library/react';

const MOCK_ON_CHANGE = jest.fn();
const MOCK_VALUES = ['one', 'two'];

describe('component | toggle', function() {
  afterEach(function() {
    cleanup();
  });

  it('renders', function() {
    const tree = renderer
      .create(<Toggle values={MOCK_VALUES} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('sends the value of next position on click', function() {
    const { container, getByTestId } = render(
      <Toggle onChange={MOCK_ON_CHANGE} values={MOCK_VALUES} />
    );

    fireEvent.click(getByTestId('toggle-button'));

    expect(container).toMatchSnapshot();
    expect(MOCK_ON_CHANGE).toHaveBeenCalledWith(MOCK_VALUES[1]);
  });

  it('sends the value of first position after clicking last', function() {
    const { container, getByTestId } = render(
      <Toggle onChange={MOCK_ON_CHANGE} values={MOCK_VALUES} />
    );

    fireEvent.click(getByTestId('toggle-button'));
    fireEvent.click(getByTestId('toggle-button'));

    expect(container).toMatchSnapshot();
    expect(MOCK_ON_CHANGE).toHaveBeenCalledWith(MOCK_VALUES[0]);
  });

  it('sends the value of next position on enter', function() {
    const { container, getByTestId } = render(
      <Toggle onChange={MOCK_ON_CHANGE} values={MOCK_VALUES} />
    );

    fireEvent.keyDown(getByTestId('toggle-button'), { key: 'Enter' });

    expect(container).toMatchSnapshot();
    expect(MOCK_ON_CHANGE).toHaveBeenCalledWith(MOCK_VALUES[1]);
  });

  it('sends the value of next position on space', function() {
    const { container, getByTestId } = render(
      <Toggle onChange={MOCK_ON_CHANGE} values={MOCK_VALUES} />
    );

    fireEvent.keyDown(getByTestId('toggle-button'), { key: 'Space' });

    expect(container).toMatchSnapshot();
    expect(MOCK_ON_CHANGE).toHaveBeenCalledWith(MOCK_VALUES[1]);
  });
});
