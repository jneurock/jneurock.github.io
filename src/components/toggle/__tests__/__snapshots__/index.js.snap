// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component | toggle renders 1`] = `
<div
  aria-hidden={true}
  className="toggle"
>
  <section
    className="toggle__control"
  >
    <div
      className="toggle__bar"
    />
    <button
      className="toggle__switch"
      data-testid="toggle-button"
      data-toggle-position={0}
      onClick={[Function]}
      onKeyDown={[Function]}
    />
  </section>
</div>
`;

exports[`component | toggle sends the value of first position after clicking last 1`] = `
<div>
  <div
    aria-hidden="true"
    class="toggle"
  >
    <section
      class="toggle__control"
    >
      <div
        class="toggle__bar"
      />
      <button
        class="toggle__switch"
        data-testid="toggle-button"
        data-toggle-position="0"
      />
    </section>
  </div>
</div>
`;

exports[`component | toggle sends the value of next position on click 1`] = `
<div>
  <div
    aria-hidden="true"
    class="toggle"
  >
    <section
      class="toggle__control"
    >
      <div
        class="toggle__bar"
      />
      <button
        class="toggle__switch"
        data-testid="toggle-button"
        data-toggle-position="1"
      />
    </section>
  </div>
</div>
`;

exports[`component | toggle sends the value of next position on enter 1`] = `
<div>
  <div
    aria-hidden="true"
    class="toggle"
  >
    <section
      class="toggle__control"
    >
      <div
        class="toggle__bar"
      />
      <button
        class="toggle__switch"
        data-testid="toggle-button"
        data-toggle-position="0"
      />
    </section>
  </div>
</div>
`;

exports[`component | toggle sends the value of next position on space 1`] = `
<div>
  <div
    aria-hidden="true"
    class="toggle"
  >
    <section
      class="toggle__control"
    >
      <div
        class="toggle__bar"
      />
      <button
        class="toggle__switch"
        data-testid="toggle-button"
        data-toggle-position="0"
      />
    </section>
  </div>
</div>
`;
